src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -5476,7 +5476,58 @@ LogicalResult ONNXUniqueOp::inferShapes(
 
 LogicalResult ONNXUpsampleOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
-  return emitError(NOT_IMPLEMENTED_MESSAGE);
+  // Sanity checks on input data argument
+  if (!X().getType().isa<RankedTensorType>()) {
+    return success();
+  }
+  if (!scales().getType().isa<RankedTensorType>()) {
+    return success();
+  }
+
+  auto inputTy = X().getType().cast<RankedTensorType>();
+  int32_t inputRank = inputTy.getShape().size();
+
+  // Sanity checks on scale argument
+  auto scalesTy = scales().getType().cast<RankedTensorType>();
+  if (scalesTy.getShape().size() != 1) {
+    return emitError("Scales tensor must be rank-1");
+  }
+  if (scalesTy.getShape()[0] != inputRank) {
+    return emitError("Input tensor rank doesn't match scales tensor shape");
+  }
+
+  SmallVector<int64_t, 4> outputDims(inputRank, -1);
+
+  // Extract the scale values
+  auto scalesConstOp = getONNXConstantOp(scales());
+  if (!scalesConstOp) {
+    return success();
+  }
+  auto valueAttr = scalesConstOp.valueAttr().dyn_cast<DenseElementsAttr>();
+  if (!valueAttr) {
+    return emitError("Scales constant is not a DenseElementsAttr");
+  }
+  int scaleIdx = 0;
+  // Why are the scale values float's?
+  for (auto it = valueAttr.getValues<FloatAttr>().begin();
```

Comment:
It is possible that scale is the result of some ratio, which is a float.

Reply:
Ahh yes that makes sense. Thanks for the clarification